wordpress-mobile/WordPress-iOS
# Update domains dashboard after purchase

IMPORTANT NOTE: I'm seeing build errors in CI which I'll check out tomorrow - I'll probably need to re-download Xcode 12.

Fixes https://github.com/wordpress-mobile/WordPress-iOS/issues/17194

Also:

- Implements refreshing the domains when the blog is synchronized.
- Addresses a sorting issue with the domains in the dashboard.

## To test:

I recommend to use the Store Sandbox to test.

### Test 1: Blog domains refreshing

- Launch the app.
- Select your test blog in My Site and stay at the blog details.
- Get a domain through another device or using the webpage.
- Go back to your initial testing device and pull down to refresh.
- Open the Domains section and make sure the new domain is there.

### Test 2: Buy a domain 

- Buy a domain through either:
    - Blog Details > Register a Domain, or
    - Blog Details > Domains > Claim your free domain
- Make sure the flow works fine and the domain dashboard (or blog details URL) shows the new domain as primary.

### Test 3: Buy a domain using the webview

- Buy a domain through: Blog Details > Domains > Add a domain
- Make sure the flow works fine and the domain dashboard shows the new domain.

## Regression Notes

1. Potential unintended areas of impact

New flows, still unreleased.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:
